lib/config_cat.ex

```diff
@@ -0,0 +1,187 @@
+defmodule ConfigCat do
+  use GenServer
+
+  alias ConfigCat.{FetchPolicy, Rollout}
+  alias HTTPoison.Response
+
+  require Logger
+
+  @base_url "https://cdn.configcat.com"
+  @base_path "configuration-files"
+  @config_filename "config_v4.json"
+
+  def start_link(sdk_key, options \\ [])
+
+  def start_link(nil, _options), do: {:error, :missing_sdk_key}
+
+  def start_link(sdk_key, options) do
+    with {name, options} <- Keyword.pop(options, :name, __MODULE__),
+         {initial_config, options} <- Keyword.pop(options, :initial_config) do
+      initial_state = %{
+        config: initial_config,
+        etag: nil,
+        last_update: nil,
+        options: Keyword.merge(default_options(), options),
+        sdk_key: sdk_key
+      }
+
+      GenServer.start_link(__MODULE__, initial_state, name: name)
+    end
+  end
+
+  defp default_options, do: [api: ConfigCat.API, fetch_policy: FetchPolicy.auto()]
+
+  def get_value(key, default_value, user_or_options \\ []) do
```

Comment:
There is an other get function called get_variation_id, which main purpose is to provide a variation_id which can be sent e.g. to an analytics provider. This feature is currently behind a feature flag in ConfigCat, but it will be turned on soon. You will be able to get the variation_id from the dasboard for a rule/percentage item/default value and match it with your analytics tool.

Reply:
Extracted to #20